# Replaces OpenFOAM third-party libraries with EASYBUILD variants.
# Uses the OpenFOAM prefs mechanism and the FOAM_CONFIG_ETC variable
# to define the preferred settings without patching the original files
#
# Authors: Mark Olesen <Mark.Olesen@esi-group.com>
#
# ChangeLog:
# - v2312 - activate METIS, KaHIP and readline support in etc/config.sh/setup
#           author: Jiri Furst <jiri.furst@gmail.com>
# - v2406 - set MPFR_ARCH_PATH and GMP_ARCH_PATH to CGAL config
#           author: Jiri Furst <jiri.furst@gmail.com>
#
# -------------------------------------------------------------------------
--- /dev/null	2020-12-14 09:05:45.272769166 +0100
+++ OpenFOAM-v2012/etc/prefs.sh	2020-12-14 10:02:26.488430802 +0100
@@ -0,0 +1,7 @@
+##Easybuild## settings -*- sh -*-
+
+export FOAM_CONFIG_ETC="etc/easybuild"
+
+export WM_MPLIB=EASYBUILDMPI
+
+##Easybuild##
--- /dev/null	2020-12-14 09:05:45.272769166 +0100
+++ OpenFOAM-v2012/etc/easybuild/config.sh/CGAL	2020-12-14 10:10:55.991841204 +0100
@@ -0,0 +1,8 @@
+##Easybuild## settings -*- sh -*-
+
+export MPFR_ARCH_PATH="$EBROOTMPFR"
+export GMP_ARCH_PATH="$EBROOTGMP"
+export BOOST_ARCH_PATH="$EBROOTBOOST"
+export CGAL_ARCH_PATH="$EBROOTCGAL"
+
+##Easybuild##
--- /dev/null	2020-12-14 09:05:45.272769166 +0100
+++ OpenFOAM-v2012/etc/easybuild/config.sh/FFTW	2020-12-14 10:10:53.735843322 +0100
@@ -0,0 +1,5 @@
+##Easybuild## settings -*- sh -*-
+
+export FFTW_ARCH_PATH="$EBROOTFFTW"
+
+##EasyBuild##
--- /dev/null	2020-12-14 09:05:45.272769166 +0100
+++ OpenFOAM-v2012/etc/easybuild/config.sh/metis	2020-12-11 21:23:28.774934024 +0100
@@ -0,0 +1,6 @@
+##Easybuild## settings -*- sh -*-
+
+METIS_VERSION="metis-$EBVERSIONMETIS"
+[ -d "$METIS_ARCH_PATH" ] || METIS_ARCH_PATH="$EBROOTMETIS"
+
+##Easybuild##
--- /dev/null   2022-12-01 18:21:35.103878336 +0100
+++ OpenFOAM-v2206/etc/easybuild/config.sh/kahip       2022-12-12 20:24:07.538408981 +0100
@@ -0,0 +1,6 @@
+##Easybuild## settings -*- sh -*-
+
+KAHIP_VERSION="kahip-$EBVERSIONKAHIP"
+export KAHIP_ARCH_PATH="$EBROOTKAHIP"
+
+##Easybuild##
--- /dev/null	2020-12-14 09:05:45.272769166 +0100
+++ OpenFOAM-v2012/etc/easybuild/config.sh/readline	2020-12-11 21:23:22.534951043 +0100
@@ -0,0 +1,5 @@
+##Easybuild## settings -*- sh -*-
+
+export READLINE_ARCH_PATH="$EBROOTLIBREADLINE"
+
+##Easybuild##
--- /dev/null	2020-12-14 09:05:45.272769166 +0100
+++ OpenFOAM-v2012/etc/easybuild/config.sh/scotch	2020-12-11 21:23:17.586964539 +0100
@@ -0,0 +1,7 @@
+##Easybuild## settings -*- sh -*-
+
+export SCOTCH_VERSION="scotch_$EBVERSIONSCOTCH"
+export SCOTCH_ARCH_PATH="$EBROOTSCOTCH"
+[ -d "$SCOTCH_ARCH_PATH" ] || SCOTCH_ARCH_PATH="$SCOTCH_ROOT"
+
+##Easybuild##
--- /dev/null	2020-12-14 09:05:45.272769166 +0100
--- OpenFOAM-v2312/etc/config.sh/setup.orig    2024-01-11 09:49:24.823571481 +0100
+++ OpenFOAM-v2312/etc/config.sh/setup 2024-01-11 09:51:40.545969774 +0100
@@ -207,7 +207,9 @@
 _foamEtc -config  CGAL
 _foamEtc -config  scotch
 _foamEtc -config  FFTW
-
+_foamEtc -config  metis
+_foamEtc -config  kahip
+_foamEtc -config  readline
 
 # Finalize library paths
 # ~~~~~~~~~~~~~~~~~~~~~~

